spec/connectors/base/connector_spec.rb

```diff
@@ -0,0 +1,264 @@
+      it 'returns false' do
+        expect(subject.active_filter_config_present?).to eq(false)
+      end
+    end
+
+    context 'advanced config is present' do
+      it 'returns true' do
+        expect(subject.active_filter_config_present?).to eq(true)
+      end
+    end
+
+    context 'advanced config is nil' do
+      let(:advanced_config) {
+        nil
+      }
+
+      it_behaves_like 'advanced_config is not present'
+    end
+
+    context 'advanced config is empty' do
+      let(:advanced_config) {
+        {}
+      }
+
+      it_behaves_like 'advanced_config is not present'
+    end
+  end
+
+  context '.rules_present?' do
+    shared_examples_for 'rules are not present' do
+      it 'returns false' do
+        expect(subject.active_rules_present?).to eq(false)
+      end
+    end
+
+    context 'rules are present' do
+      it 'returns true' do
+        expect(subject.active_rules_present?).to eq(true)
+      end
+    end
+
+    context 'rules are nil' do
+      let(:active_rules) {
+        nil
+      }
+
+      it_behaves_like 'rules are not present'
+    end
+
+    context 'rules are empty' do
+      let(:active_rules) {
+        []
+      }
+
+      it_behaves_like 'rules are not present'
+    end
+  end
+
+  context '.filtering_present?' do
+    shared_examples_for 'filtering is not present' do
+      it 'returns false' do
+        expect(subject.filtering_present?).to eq(false)
+      end
+    end
+
+    shared_examples_for 'filtering is present' do
+      it 'returns true' do
+        expect(subject.filtering_present?).to eq(true)
+      end
+    end
+
+    context 'rules and advanced_config are present' do
+      it_behaves_like 'filtering is present'
+    end
+
+    context 'only rules are nil' do
+      let(:active_rules) {
+        nil
+      }
+
+      it_behaves_like 'filtering is present'
+    end
+
+    context 'only rules are empty' do
+      let(:active_rules) {
+        []
+      }
+
+      it_behaves_like 'filtering is present'
+    end
+
+    context 'advanced_config is empty' do
+      let(:advanced_config) {
+        {}
+      }
+
+      it_behaves_like 'filtering is present'
+    end
+
+    context 'advanced_config is nil' do
+      let(:advanced_config) {
+        nil
+      }
+
+      it_behaves_like 'filtering is present'
+    end
+
+    context 'rules are empty and advanced_config is empty' do
+      let(:active_rules) {
+        []
+      }
+
+      let(:advanced_config) {
+        {}
+      }
+
+      it_behaves_like 'filtering is not present'
+    end
+
+    context 'rules are nil and advanced_config is nil' do
+      let(:active_rules) {
+        nil
+      }
+
+      let(:advanced_config) {
+        nil
+      }
+
+      it_behaves_like 'filtering is not present'
+    end
+
+    context '.initialize' do
+      context 'three rules are present' do
+        let(:active_rules) {
+          [
+          { :name => 'rule one' },
+          { :name => 'rule two' },
+          { :name => 'rule three' },
```

Comment:
Just as a note, rules don't have `:name` properties as we've defined them. I think this test is fine as it is until we get to https://github.com/elastic/enterprise-search-team/issues/3168, but wanted to make sure we're on the same page that this isn't really representative right now.